honeycombio/honeycomb-opentelemetry-node
# maint: update opentelemetry dependencies

## Which problem is this PR solving?

When installing in a new app, version conflicts caused the instrumentation to not work properly.

More investigation may be needed to help reduce conflicts in the future, but right now it seems that the most reliable usage is to use more current versions and to have the same versions in our dependencies and in end user apps, especially for the `api` and `auto-instrumentations-node` packages.

## Short description of the changes

```text
    "@opentelemetry/api": "^1.2.0", --> "^1.3.0"
    "@opentelemetry/exporter-trace-otlp-grpc": "^0.33.0", --> "^0.34.0"
    "@opentelemetry/exporter-trace-otlp-proto": "^0.33.0", --> "^0.34.0"
    "@opentelemetry/resources": "^1.7.0", --> "^1.8.0"
    "@opentelemetry/sdk-node": "^0.33.0", --> "^0.34.0"
    "@opentelemetry/sdk-trace-base": "^1.7.0",--> "^1.8.0"
```

Also, notes were added into `DEVELOPING.md` about how to package as a tarball for easier local development in other projects. 

## How to verify that this has the expected result

install `@honeycombio/opentelemetry-node`, `@opentelemetry/api`, and `@opentelemetry/auto-instrumentations-node`, ensuring versions `v0.1.1-beta`, `1.3.0`, and `0.34.0` are installed respectively. Use as instructed in the docs, and see both auto- and manual instrumentation. Also smoke tests should work.

Co-authored-by: Purvi Kanal <purvikanal@honeycomb.io>
